lib/vagrant-libvirt/config.rb

```diff
@@ -538,6 +538,15 @@ def _generate_uri
       end
 
       def finalize!
+        @uri = ENV['VAGRANT_LIBVIRT_URI'] if (@uri == UNSET_VALUE && ENV['VAGRANT_LIBVIRT_URI'])
+        if (@uri && @uri != UNSET_VALUE)
+          uri = URI(@uri)
+          @driver = u.scheme.sub('+ssh','') if @driver == UNSET_VALUE
```

Comment:
Btw, this is probably wrong, as driver controls what type of virtualization should be used, qemu, qemu accelerated or kvm, but that is not inferable from the uri.

See https://libvirt.org/uri.html#URI_qemu for more info, and the `_generate_uri` method above in that usage of either kvm or qemu as the driver will result in a `qemu` scheme for the connection.

Course there is also a typo here in that `u.scheme` should be `uri.scheme`.